cmd/autotestgridgenerator/main.go

```diff
@@ -0,0 +1,184 @@
+	fs.StringVar(&o.assign, "assign", githubTeam, "The github username or group name to assign the created pull request to.")
+	fs.StringVar(&o.testgridConfigDir, "testgrid-config", "../../../../kubernetes/test-infra/config/testgrids/openshift", "The directory where the testgrid output will be stored")
+	fs.StringVar(&o.releaseConfigDir, "release-config", "../../../release/core-services/release-controller/_releases", "The directory of release config files")
+	fs.StringVar(&o.prowJobsDir, "prow-jobs-dir", "../../../release/ci-operator/jobs", "The directory where prow-job configs are stored")
+	fs.StringVar(&o.allowList, "allow-list", "../../../release/core-services/testgrid-config-generator/_allow-list.yaml", "File containing release-type information to override the defaults")
+
+	fs.BoolVar(&o.selfApprove, "self-approve", false, "Self-approve the PR by adding the `approved` and `lgtm` labels. Requires write permissions on the repo.")
+	o.AddFlags(fs)
+	o.AllowAnonymous = true
+	if err := fs.Parse(os.Args[1:]); err != nil {
+		logrus.WithError(err).Errorf("cannot parse args: '%s'", os.Args[1:])
+	}
+	return o
+}
+
+func validateOptions(o options) error {
+	if o.githubLogin == "" {
+		return fmt.Errorf("--github-login cannot be empty string")
+	}
+	if (o.gitEmail == "") != (o.gitName == "") {
+		return fmt.Errorf("--git-name and --git-email must be specified together")
+	}
+	if o.targetDir == "" {
+		return fmt.Errorf("--target-dir is mandatory")
+	}
+	if o.assign == "" {
+		return fmt.Errorf("--assign is mandatory")
+	}
+	return o.GitHubOptions.Validate(false)
+}
+
+func runAndCommitIfNeeded(stdout, stderr io.Writer, author, cmd string, args []string) (bool, error) {
+	fullCommand := fmt.Sprintf("%s %s", filepath.Base(cmd), strings.Join(args, " "))
+
+	logrus.Infof("Running: %s", fullCommand)
+	if err := bumper.Call(stdout, stderr, cmd, args...); err != nil {
+		return false, fmt.Errorf("failed to run %s: %w", fullCommand, err)
+	}
+
+	changed, err := bumper.HasChanges()
+	if err != nil {
+		return false, fmt.Errorf("error occurred when checking changes: %w", err)
+	}
+
+	if !changed {
+		logrus.WithField("command", fullCommand).Info("No changes to commit")
+		return false, nil
+	}
+
+	gitCmd := "git"
+	if err := bumper.Call(stdout, stderr, gitCmd, []string{"add", "."}...); err != nil {
+		return false, fmt.Errorf("failed to 'git add .': %w", err)
+	}
+
+	commitArgs := []string{"commit", "-m", fullCommand, "--author", author}
+	if err := bumper.Call(stdout, stderr, gitCmd, commitArgs...); err != nil {
+		return false, fmt.Errorf("fail to %s %s: %w", gitCmd, strings.Join(commitArgs, " "), err)
+	}
+
+	return true, nil
+}
+
+func main() {
+	o := parseOptions()
+	if err := validateOptions(o); err != nil {
+		logrus.WithError(err).Fatal("Invalid arguments.")
+	}
+	// targetDir needs to be the test-infra directory
+	logrus.Infof("Changing working directory to '%s' ...", o.targetDir)
+	if err := os.Chdir(o.targetDir); err != nil {
```

Comment:
IMO you cannot simply change the working dir like this. The users expect to provide inputs (parameters) relative to where they are, and when you `chdir`, the paths they supplied (token path, allow list, everything) are now invalid when they are relative.

Reply:
I am changing the working directory so that the git operations would work properly ie the git add of the changed files happen in the necessary repository. I can instead add something like --work-tree and --add-dir.
Is that what you meant?

Reply:
I think you can use just `-C`, but yes. I think that's preferable to remembering to make all user input paths absolute first so you can `chdir`.

Reply:
Ok.. I'll switch to -C. Is this just a convenient way to change the directory or is there some additional benefit as well?